ext/opcache/tests/opt/infer_003.phpt

```diff
@@ -0,0 +1,90 @@
+--TEST--
+Infer 003: '==='/'!==' optimizations
+--INI--
+opcache.enable=1
+opcache.enable_cli=1
+opcache.optimization_level=-1
+opcache.opt_debug_level=0x20000
+opcache.preload=
+--SKIPIF--
+<?php require_once('skipif.inc'); ?>
+--FILE--
+<?php
+class X {
+    public static function create_string(int $i): string {
+        return "a$i";
+    }
+    public static function foo(int $a) {
+        for ($i = 0; $i < 2; $i++) {
+            if (self::create_string($i) === $a) {
+                echo "Impossible\n";
+            }
+            if (self::create_string($i) === []) {
+                echo "Impossible\n";
+            }
+            if (self::create_string($i) === self::create_string(1)) {
+                echo "Success for $i\n";
+            }
+        }
+    }
+}
+X::foo(1);
+?>
+--EXPECTF--
+$_main:
+     ; (lines=4, args=0, vars=0, tmps=0)
+     ; (after optimizer)
+     ; %s:1-22
+0000 INIT_STATIC_METHOD_CALL 1 string("X") string("foo")
+0001 SEND_VAL int(1) 1
+0002 DO_UCALL
+0003 RETURN int(1)
+
+X::create_string:
+     ; (lines=3, args=1, vars=1, tmps=1)
+     ; (after optimizer)
+     ; %s:3-5
+0000 CV0($i) = RECV 1
+0001 T1 = FAST_CONCAT string("a") CV0($i)
+0002 RETURN T1
+
+X::foo:
+     ; (lines=29, args=1, vars=2, tmps=3)
+     ; (after optimizer)
+     ; %s:6-18
+0000 CV0($a) = RECV 1
+0001 CV1($i) = QM_ASSIGN int(0)
+0002 JMP 0026
+0003 INIT_STATIC_METHOD_CALL 1 (self) (exception) string("create_string")
+0004 SEND_VAR CV1($i) 1
+0005 DO_UCALL
+0006 JMP 0008
```

Comment:
The behavior change in `ext/opcache/tests/opt/sccp_026.phpt` is similar to this - the opcode 0007 is unreachable and unconditionally skipped over but dead code detection fails to eliminate it.